roles/lib_openshift/src/test/unit/test_oc_scale.py

```diff
@@ -1,256 +1,110 @@
-        ''' Testing binary lookup fallback '''
-
-        mock_env_get.side_effect = lambda _v, _d: ''
-
-        mock_shutil_which.side_effect = lambda _f, path=None: None
-
-        self.assertEqual(locate_oc_binary(), 'oc')
-
-    @unittest.skipIf(six.PY2, 'py3 test only')
-    @mock.patch('shutil.which')
-    @mock.patch('os.environ.get')
-    def test_binary_lookup_in_path_py3(self, mock_env_get, mock_shutil_which):
-        ''' Testing binary lookup in path '''
-
-        oc_bin = '/usr/bin/oc'
-
-        mock_env_get.side_effect = lambda _v, _d: '/bin:/usr/bin'
-
-        mock_shutil_which.side_effect = lambda _f, path=None: oc_bin
-
-        self.assertEqual(locate_oc_binary(), oc_bin)
-
-    @unittest.skipIf(six.PY2, 'py3 test only')
-    @mock.patch('shutil.which')
-    @mock.patch('os.environ.get')
-    def test_binary_lookup_in_usr_local_py3(self, mock_env_get, mock_shutil_which):
-        ''' Testing binary lookup in /usr/local/bin '''
-
-        oc_bin = '/usr/local/bin/oc'
-
-        mock_env_get.side_effect = lambda _v, _d: '/bin:/usr/bin'
-
-        mock_shutil_which.side_effect = lambda _f, path=None: oc_bin
-
-        self.assertEqual(locate_oc_binary(), oc_bin)
-
-    @unittest.skipIf(six.PY2, 'py3 test only')
-    @mock.patch('shutil.which')
-    @mock.patch('os.environ.get')
-    def test_binary_lookup_in_home_py3(self, mock_env_get, mock_shutil_which):
-        ''' Testing binary lookup in ~/bin '''
-
-        oc_bin = os.path.expanduser('~/bin/oc')
-
-        mock_env_get.side_effect = lambda _v, _d: '/bin:/usr/bin'
-
-        mock_shutil_which.side_effect = lambda _f, path=None: oc_bin
-
-        self.assertEqual(locate_oc_binary(), oc_bin)
+import json
+
+from lib_openshift.library import oc_scale
+
+MODULE_UNDER_TEST = oc_scale
+CLASS_UNDER_TEST = oc_scale.OCScale
+
+
+def test_state_list(mock_run_cmd):
+    ''' Testing a get '''
+    params = {'name': 'router',
+              'namespace': 'default',
+              'replicas': 2,
+              'state': 'list',
+              'kind': 'dc',
+              'kubeconfig': '/etc/origin/master/admin.kubeconfig',
+              'debug': False}
+
+    dc = {
+        "kind": "DeploymentConfig",
+        "apiVersion": "v1",
+        "metadata": {
+            "name": "router",
+            "namespace": "default",
+            "selfLink": "/oapi/v1/namespaces/default/deploymentconfigs/router",
+            "uid": "a441eedc-e1ae-11e6-a2d5-0e6967f34d42",
+            "resourceVersion": "6558",
+            "generation": 8,
+            "creationTimestamp": "2017-01-23T20:58:07Z",
+            "labels": {
+                "router": "router"
+            }
+        },
+        "spec": {
+            "replicas": 2,
+        }
+    }
+
+    mock_run_cmd.side_effect = [
+        (0, json.dumps(dc), '')
+    ]
+
+    results = CLASS_UNDER_TEST.run_ansible(params, False)
+
+    assert results['changed'] is False
```

Comment:
I tried using `not` and I think being a bit more explicit with `is False` reads a bit better. It also prevents explicit type conversion. I think I might update the positive changed tests to be more explicit for that reason as well.